File: frontend/src/types/publicStats.ts
import { RiskLevelStats } from "@/types/admin";

export interface DailyTrendPoint {
  date: string;
  total: number;
  safe: number;
  caution: number;
  danger: number;
  unverified: number;
}

export interface TopClaimedOrg {
  org: string;
  count: number;
}

export interface RecentAlert {
  claimed_org: string | null;
  claim_type: string | null;
  risk_level: string | null;
  created_at: string;
}

export interface PublicStatsResponse {
  total_count: number;
  risk_level_stats: RiskLevelStats;
  daily_trend: DailyTrendPoint[];
  top_claimed_orgs: TopClaimedOrg[];
  recent_alerts: RecentAlert[];
}

File: backend/app/schemas/public_stats.py
from datetime import date, datetime

from pydantic import BaseModel

from app.schemas.admin import RiskLevelStats


class DailyTrendPoint(BaseModel):
    date: date
    total: int
    safe: int
    caution: int
    danger: int
    unverified: int


class TopClaimedOrg(BaseModel):
    org: str
    count: int


class RecentAlert(BaseModel):
    claimed_org: str | None
    claim_type: str | None
    risk_level: str | None
    created_at: datetime


class PublicStatsResponse(BaseModel):
    total_count: int
    risk_level_stats: RiskLevelStats
    daily_trend: list[DailyTrendPoint]
    top_claimed_orgs: list[TopClaimedOrg]
    recent_alerts: list[RecentAlert]
